frontend/public/components/alert.tsx

```diff
@@ -0,0 +1,469 @@
+  props: AlertsPageProps;
+  defaultNameFilter: string;
+
+  constructor (props) {
+    super(props);
+    this.applyTextFilter = this.applyTextFilter.bind(this);
+  }
+
+  applyTextFilter (e) {
+    const v = e.target.value;
+    store.dispatch(k8sActions.filterList(reduxID, nameFilterID, v));
+
+    const params = new URLSearchParams(window.location.search);
+    if (v) {
+      params.set(nameFilterID, v);
+    } else {
+      params.delete(nameFilterID);
+    }
+    const url = new URL(window.location.href);
+    history.replace(`${url.pathname}?${params.toString()}${url.hash}`);
+  }
+
+  componentWillMount () {
+    const params = new URLSearchParams(window.location.search);
+
+    // Ensure the current name filter value matches the name filter GET param
+    this.defaultNameFilter = params.get(nameFilterID);
+    store.dispatch(k8sActions.filterList(reduxID, nameFilterID, this.defaultNameFilter));
+
+    if (!params.get('sortBy')) {
+      // Sort by rule name by default
+      store.dispatch(UIActions.sortList(reduxID, 'name', undefined, 'asc', 'Name'));
+    }
+  }
+
+  render () {
+    const {filters, loaded, loadError, rules} = this.props;
+
+    return <React.Fragment>
+      <NavTitle title="Monitoring Alerts" />
+      <div className="co-m-pane__filter-bar">
+        <div className="co-m-pane__filter-bar-group">
+          <AlertsPageDescription />
+          <div className="co-m-pane__filter-bar-group--filter">
+            <TextFilter defaultValue={this.defaultNameFilter} label="Alerts by name" onChange={this.applyTextFilter} />
+          </div>
+        </div>
+      </div>
+      <div className="co-m-pane__body">
+        <div className="row">
+          <CheckBoxes
+            items={rowFilter.items}
+            numbers={_.countBy(rules, rowFilter.reducer)}
+            reduxIDs={[reduxID]}
+            selected={rowFilter.selected}
+            type={rowFilter.type}
+          />
+        </div>
+        <div className="row">
+          <div className="col-xs-12">
+            <List
+              Header={AlertHeader}
+              Row={Row}
+              data={rules}
+              filters={filters}
+              loadError={loadError}
+              loaded={loaded}
+              reduxID={reduxID}
+            />
+          </div>
+        </div>
+      </div>
+    </React.Fragment>;
+  }
+});
+export const AlertsPage = connectPage(AlertsPage_);
+
+/*  eslint-disable no-unused-vars  */
+type Alert = {
+  activeAt: string;
+  annotations: any;
+  labels: {[key: string]: string};
+  state: string;
+  value: any,
+};
+type Rule = {
+  alerts: Array<Alert>;
+  annotations: any;
+  duration: number;
+  labels: {[key: string]: string};
+  query: string;
+};
+type StateProps = {
+  state: string;
+};
+type ReduxData = {
+  loaded: boolean;
+  loadError?: string;
+  rules: Array<Rule>;
+};
+type AlertsPageWrapperProps = {
+  Page: React.ComponentType<any>;
+};
+type AlertsDetailsPageProps = {
```

Comment:
Export theses types so they can be used in unit tests.

Reply:
👍